src/session/sessionStore.ts

```diff
@@ -0,0 +1,36 @@
+import { AppSessionCookieStoreFactory, AppSessionStore } from './types'
+import { refreshStoredAppSession } from './refreshStoredAppSession'
+import { cookieAdapter } from '../session-adapters/cookieAdapter'
+import { createInternalAdapter } from '../session-adapters/internalAdapter'
+
+export const getSessionStore: AppSessionCookieStoreFactory =
+  (params) =>
+  (requestParams): AppSessionStore => {
+    const adapter = createInternalAdapter({
+      params,
+      req: requestParams.req,
+      res: requestParams.res,
+      adapter: cookieAdapter,
+    })
+
+    return {
+      get: async ({ spaceId, userId }) => {
+        const session = await adapter.getSession({
+          spaceId: String(spaceId),
+          userId: String(userId),
+        })
+        return await refreshStoredAppSession(params, adapter, session)
```

Comment:
I am thinking that if this moved to internalAdapter then maybe there would be no need for us to even have the init_oauth flag? WDYT? and we coul even remove the middleware logic from the nuxt-base

Reply:
Update: Actually no I think this way it should already work even if we remove the nuxt-base init_oauth conditional, because when we get session it will first check it and if its stale then it will refresh